app/src/backend/ResultsService.ts

```diff
@@ -426,6 +436,36 @@ function convertEmissionsToStrings(
   };
 }
 
+async function calculateThirdPartyEmissionsByScope(
+  inventoryId: string,
+): Promise<ActivityDataByScope[]> {
+  const inventoryValues = await db.models.InventoryValue.findAll({
+    where: { inventoryId },
+  });
+  const scopes = inventoryValues.map((value) => {
+    const scopeName = value.gpcReferenceNumber?.split(".").slice(-1)[0];
+
+    if (!scopeName) {
+      throw new createHttpError.InternalServerError(
+        "Scope name not found for inventory value " + value.id,
+      );
+    }
+
+    const co2eq = value.co2eq ?? 0n;
+    const totalEmissions = bigIntToDecimal(co2eq);
+    return {
+      activityTitle: "third-party", // TODO use data source name or SubCategory name?
+      scopes: {
+        [scopeName]: bigIntToDecimal(co2eq),
+      },
+      totalEmissions,
+      percentage: 100,
+    };
```

Comment:
Shouldn't this be limited to inventory values with a data source id? Especially if the goal is to get a scope breakdown of thirdparty emissions

Reply:
Good point, done :+1: I'm also using the dataset name instead of `third-party` now.